xbmc/addons/AddonDll.cpp

```diff
@@ -425,5 +523,233 @@ bool CAddonDll::CheckAPIVersion(int type)
+      logLevel = LOGWARNING;
+      break;
+    case ADDON_LOG_NOTICE:
+      logLevel = LOGNOTICE;
+      break;
+    case ADDON_LOG_INFO:
+      logLevel = LOGINFO;
+      break;
+    case ADDON_LOG_DEBUG:
+      logLevel = LOGDEBUG;
+      break;
+    default:
+      break;
+  }
+
+  CLog::Log(logLevel, "AddOnLog: %s: %s", addon->Name().c_str(), strMessage);
+}
+
+bool CAddonDll::get_setting(void* kodiBase, const char* settingName, void* settingValue)
+{ 
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (addon == nullptr || settingName == nullptr || settingValue == nullptr)
+  {
+    CLog::Log(LOGERROR, "kodi::General::%s - invalid data (addon='%p', settingName='%p', settingValue='%p')",
+                                        __FUNCTION__, addon, settingName, settingValue);
+
+    return false;
+  }
+
+  CLog::Log(LOGDEBUG, "kodi::General::%s - add-on '%s' requests setting '%s'",
+                        __FUNCTION__, addon->Name().c_str(), settingName);
+
+  if (strcasecmp(settingName, "__addonpath__") == 0)
+  {
+    strcpy((char*) settingValue, addon->Path().c_str());
+    return true;
+  }
+
+  if (!addon->ReloadSettings())
+  {
+    CLog::Log(LOGERROR, "kodi::General::%s - could't get settings for add-on '%s'", __FUNCTION__, addon->Name().c_str());
+    return false;
+  }
+
+  const TiXmlElement *category = addon->GetSettingsXML()->FirstChildElement("category");
+  if (!category) // add a default one...
+    category = addon->GetSettingsXML();
+
+  while (category)
+  {
+    const TiXmlElement *setting = category->FirstChildElement("setting");
+    while (setting)
+    {
+      const std::string   id = XMLUtils::GetAttribute(setting, "id");
+      const std::string type = XMLUtils::GetAttribute(setting, "type");
+
+      if (id == settingName && !type.empty())
+      {
+        if (type == "text"     || type == "ipaddress" ||
+            type == "folder"   || type == "action"    ||
+            type == "music"    || type == "pictures"  ||
+            type == "programs" || type == "fileenum"  ||
+            type == "file"     || type == "labelenum")
+        {
+          strcpy((char*) settingValue, addon->GetSetting(id).c_str());
+          return true;
+        }
+        else if (type == "number" || type == "enum")
+        {
+          *(int*) settingValue = (int) atoi(addon->GetSetting(id).c_str());
+          return true;
+        }
+        else if (type == "bool")
+        {
+          *(bool*) settingValue = (bool) (addon->GetSetting(id) == "true" ? true : false);
+          return true;
+        }
+        else if (type == "slider")
+        {
+          const char *option = setting->Attribute("option");
+          if (option && strcmpi(option, "int") == 0)
+          {
+            *(int*) settingValue = (int) atoi(addon->GetSetting(id).c_str());
+            return true;
+          }
+          else
+          {
+            *(float*) settingValue = (float) atof(addon->GetSetting(id).c_str());
```

Comment:
Want to leave them currently like it is, also thought to change them but with the coming https://github.com/xbmc/xbmc/pull/12125 I want to hold differences to them on "xbmc/addons/interfaces/Addon/AddonCallbacksAddon.cpp" so small as possible.

OK for you?